src/main/java/de/tum/in/www1/artemis/service/BitbucketService.java

```diff
@@ -107,16 +107,24 @@ public URL getRepositoryWebUrl(Participation participation) {
 
     private String getProjectKeyFromUrl(URL repositoryUrl) {
         // https://ga42xab@repobruegge.in.tum.de/scm/EIST2016RME/RMEXERCISE-ga42xab.git
-        return repositoryUrl.getFile().split("/")[2];
+        String[] urlParts = repositoryUrl.getFile().split("/");
+        if (urlParts.length > 2) {
+            return urlParts[2];
+        }
+        return "";
```

Comment:
what happens in the other code parts, in case we return "" here?
Of course we avoid an exception, but do we have any side effects when using an empty string? (sometimes an exception might be better)

Reply:
In theory thats right, however I can only imagine of having an invalid URL when creating the exercise (here the repositoryUrlIsValid-method will return false because BitBucket cannot find the repository if either the project slug or repo name is empty). In any other case, we assume we have a correct URL (as we assume we only create correct URLs).

Reply:
@krusche I just thought about it, an exception problably really is better. I will change it.